### PR TITLE
UI - Scale VM - Fix compute offering selection not working

### PR DESCRIPTION
**Pull request by utchoang**
### Description

Fixes https://github.com/apache/cloudstack/issues/5391

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by rhtyd** [comment]
@blueorangutan ui

**Comment by davidjumani** [comment]
@utchoang I think another issue is the cpu speed slider is not being shown. Can you look into that?

**Comment by kricud** [comment]
@utchoang There is such field in legacy UI. It’s needed as its mandatory in offering. It would be grate that this field is prefilled, and admin can control default/prefilled value. Now for every new user trying to use this ticket is created with question what I must enter in CPU (in MHz) field.

Screenshot 2021-08-31 152904

**Comment by utchoang** [comment]
@davidjumani In my opinion, it would make more sense to not display the slider here anymore. What do you think?

**Comment by davidjumani** [comment]
@utchoang Yes, I think the slider is pointless unless it is a constrained offering (show it only for values within a range of cpu / memory and let the speed be a regular field) cc @rhtyd

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
@blueorangutan ui

File: ui/src/views/compute/wizard/ComputeSelection.vue [hunk 1]
```diff
@@ -25,7 +25,7 @@
             :validate-status="errors.cpu.status"
             :help="errors.cpu.message">
             <a-row :gutter="12">
-              <a-col :md="10" :lg="10" v-show="isConstrained">
+              <a-col :md="10" :lg="10" v-show="isConstrained && maxCpu && isNaN(maxCpu)">
```
Comment (shwstppr) [inline ui/src/views/compute/wizard/ComputeSelection.vue:28]: @utchoang is this show condition correct here. For memory you have used reversed for isNaN which probably makes sense

Comment (utchoang) [reply ui/src/views/compute/wizard/ComputeSelection.vue:28]: @shwstppr Yes. I'm wrong. Thanks

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by kricud** [comment]
@utchoang 
In Legacy UI if you want to use custom offering you need to enter/fill 3 values 
CPU (in MHz), # of CPU Cores, Memory (in MB) they are mandatory.

In New UI you can enter only 2 values CPU Cores, Memory (in MB).
If you want to create service offering manually or dynamically you need to specify 
CPU (in MHz) in both cases its mandatory field.

New UI is missing CPU (in MHz).

My two cents:
New UI sliders in their existing form are pointless:
For RAM slider you would expect increments in 1024MB.
For CPU Cores you would expect averaged maximum of host vCPU.
For CPU MHz you would expect all host MHz options.
I would suggest remove sliders if only things they do is increase number from 1 to something.

New UI:
Screenshot 2021-09-01 122614

Legacy UI:
Screenshot 2021-09-01 122712

Static offering:
Screenshot 2021-09-01 123235

**Comment by shwstppr** [comment]
@kricud Is **Custom compute offering - A** a constrained custom offering or an unconstrained offering
In my test for this PR, UI seems to be showing correct fields for both constrained and unconstrained compute offerings.
No sliders for unconstrained custom offering
Screenshot from 2021-09-01 15-25-36
Sliders for constrained custom offering
Screenshot from 2021-09-01 15-25-53

Regarding max value for sliders, they are based on the max value set in the compute offering at the time of the creation. I don't think we need to involve host CPU/memory here.
Although the step for slider can be tweaked based on the min-max if it is not done already.

**Comment by kricud** [comment]
@shwstppr @utchoang My fail - operating with constrained offerings and raging on keyboard why its not doing what I’m expecting. Apologies for BS typing.

**Comment by kricud** [comment]
@shwstppr @utchoang Now it all came together- in our case its constrained offering that has no constrains.
It was created on 4.11.3 after migration to 4.15.1 its all bugging as it has missing mandatory values.

Name	Custom compute offering - A
ID	5adb6fde-60e6-4bd1-b8ab-d48dae8f5fa0
Description	Custom compute offering - A
Storage Type	shared
Provisioning Type	thin
Write-cache Type	
# of CPU Cores	
CPU (in MHz)	
Memory (in MB)	
Network Rate (Mb/s)	1000
Custom IOPS	
Min IOPS	N/A
Max IOPS	N/A
Hypervisor Snapshot Reserve	N/A
Disk Read Rate (BPS)	
Disk Write Rate (BPS)	
Disk Read Rate (IOPS)	
Disk Write Rate (IOPS)	
Offer HA	Yes
CPU Cap	No
Volatile	No
Deployment planner	
Planner mode	
GPU	
vGPU type	
Storage Tags	
Host Tag	
Domain	ROOT
Zone	
Created	08 Dec 2020 13:27:46

Apologies once more.

**Comment by shwstppr** [comment]
@kricud No problem.
I think the concept of constrained-unconstrained custom compute offerings got introduced in 4.14. Maybe there could be an issue with how custom compute offerings created prior to that were handled. Will try to check with a 4.11 to 4.14/4.15 upgrade.

**Review by shwstppr — APPROVED** [review]
LGTM. Checked different users of `ComputeSelection.vue` and ScaleVM form

**Review by nvazquez — APPROVED** [review]
LGTM

**Pull request merged**